astropy/table/table.py

```diff
@@ -896,24 +900,30 @@ def show_in_notebook(self, tableid=None, css=None, display_length=50,
                                             np.random.randint(1, 1e6))
 
         jsv = JSViewer(display_length=display_length)
-        html = self._base_repr_(html=True, max_width=-1, tableid=tableid,
-                                max_lines=-1, show_dtype=False,
-                                tableclass=table_class)
+
+        has_row_index = 'row_index' in self.colnames
```

Comment:
Although `row_index` is strictly more correct, I wonder if just `row` would suffice.  I'm bothered by the big chunk of wasted white space in the prime real estate on the left side of the table.

I also wonder if we could play silly games of adding space characters to the left and right of `row` until it is no longer in the original table `colnames`.